.github/workflows/checkers-action.yml

```diff
@@ -0,0 +1,75 @@
+name: Checkers-Action
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'cve_bin_tool/checkers/**/*.py'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'cve_bin_tool/checkers/**/*.py'
+
+jobs:
+  run-script:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get date 
+        id: get-date 
+        run: | 
+          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT 
+          echo "yesterday=$(/bin/date -d "-1 day" -u "+%Y%m%d")" >> $GITHUB_OUTPUT 
+
+      - name: Print Cache Keys 
+        run: | 
+          echo "Today's Cache Key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}" 
+          echo "Yesterday's Cache Key: Linux-cve-bin-tool-${{ steps.get-date.outputs.yesterday }}" 
+
+      - name: Get today's cached database 
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0 
+        id: todays-cache 
+        with: 
+          path: cache 
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }} 
+
+      - name: Get yesterday's cached database if today's is not available 
+        uses: actions/cache@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0 
+        if: steps.todays-cache.outputs.cache-hit != 'true' 
+        with: 
+          path: cache 
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.yesterday }}  
+
+      - name: Install cve-bin-tool
+        if: env.sbom != 'true'
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade setuptools
+          python -m pip install --upgrade wheel
+          python -m pip install --upgrade -r dev-requirements.txt
+          python -m pip install --upgrade .
+          
+      - name: Try single CLI run of tool
+        if: env.sbom != 'true'
+        run: |
+          [[ -e cache ]] && mkdir -p .cache && mv cache ~/.cache/cve-bin-tool
+          NO_EXIT_CVE_NUM=1 python -m cve_bin_tool.cli test/assets/test-kerberos-5-1.15.1.out
+          cp -r ~/.cache/cve-bin-tool cache
+
+      - name: Get changed files in checkers directory
+        id: changed-files
+        run: |
+          files=$(git diff --name-only ${{ github.sha }} ${{ github.event.before }} | grep '^cve_bin_tool/checkers/' | xargs)
```

Comment:
This is generating an error when run and I don't know offhand what to do with it, so you're on your own for fixing it:

```
Run files=$(git diff --name-only 6abca452cae5546029a21f9b8db03a2a6ee5c822 9bf7a8e572bc4ab05248b861c5d271b48792631a | grep '^cve_bin_tool/checkers/' | xargs)
fatal: bad object 9bf7a8e572bc4ab05248b861c5d271b48792631a
```

Reply:
Yepp. I tried changing the depth of the checkout code to 0 and it fetched all the checker files. I'm not able to fetch only the changed .py files in checkers.

Reply:
We're using https://github.com/technote-space/get-diff-action on some of our other workflows; don't know if it'll be better but maybe worth a shot?

Reply:
Yeah, but one branch can have multiple commit so why not just check branches for the difference.